### DISCUSSION TITLE
Slow relationship connections

### DISCUSSION THREAD

**Discussion by JCV67** [Budibase/budibase#11912] [Help]
I am having some performance issues related to relationship pickers.

The above form takes 10s to fully load. The normal fields appear filed almost immediately but the relationship fields take 10s to fully load.

Could it be a server configuration issue?
Is it a problem related to internal database?
Should I rebuild the app in an external postgres DB?

![form](https://github.com/Budibase/budibase/assets/104722906/eaeb6d41-c8d2-4516-8d16-684c4f1bc0ff)

self hosted | 2.10.12

**Comment by melohagan** [comment]
Hey @JCV67 

The relationship pickers are being improved upon in the upcoming release, hopefully that will improve the performance for you!

**Reply by JCV67** [reply]
Great. Thank you.
